### PR TITLE
Add more instructions to bril ocaml

### PR DESCRIPTION
**Pull request by fyquah**
Specifically, the alloc, store, load, ptradd and free.

I tested this on some of the test cases that requires this instructions.

### CODE REVIEW

**Comment by sampsyo** [comment]
Looks excellent; many thanks!! @kq-li, care to take a quick look before we merge?

File: bril-ocaml/lib/instr.ml [hunk 1]
```diff
@@ -131,13 +164,30 @@ let of_json json =
...
+    `Assoc (
+      [ ("op", `String op) ]
+      @ (match args with
+          | None -> []
+          | Some args ->
+            [ "args", `List (List.map args ~f:(fun a -> `String a)) ])
+      @ (match dest with
+          | None -> []
+          | Some dest -> dest_to_json dest)
+    )
```
Comment (kq-li) [inline bril-ocaml/lib/instr.ml:189]: i'd use `List.concat` here

**Review by kq-li — APPROVED** [review]
looks good to me! might be good to run `ocamlformat` before final merge just for style consistency

File: bril-ocaml/lib/instr.ml (hunk 1 above)
Comment (fyquah) [reply bril-ocaml/lib/instr.ml]: done, force pushed.

**Comment by fyquah** [comment]
Done, i simply ran dune build @fmt and promoted the relevant changes.

btw, I made some changes to the ocaml documentation. I think the opam installation instruction is slightly out-of-date and doesn't work on newer opam versions. (https://github.com/sampsyo/bril/pull/130).

**Comment by sampsyo** [comment]
Wahoo!! Thanks, @fyquah, and thanks for the comments, @kq-li.

**Pull request merged**